Libs/vtkSegmentationCore/vtkSegmentation.h

```diff
@@ -105,7 +107,9 @@ class vtkSegmentationCore_EXPORT vtkSegmentation : public vtkObject
     /// Invoked if a representation is created or removed in the segments (e.g., created by conversion from master).
     ContainedRepresentationNamesModified,
     /// Invoked if segment IDs order is changed. Not called when a segment is added or removed.
-    SegmentsOrderModified
+    SegmentsOrderModified,
+    /// Invoked when content the data object of a representation is changed.
```

Comment:
In this case, the event was meant to indicate that the underlying vtkDataObject has been changed to a different one. It does not include situations in which the segment representation has just been modified (which are covered by RepresentationModified and MasterRepresentationModified).

Reply:
I see what's the difference now. For some reason, it is not common in VTK (or in Slicer, at the level of logic classes) to distringuish between these. I.e., if you modifiy points, it does not matter if you replaced the entire pointdata object or you modified point coordinates in  the same pointdata object. I know that in GUI classes this is an important distinction, and there are quite often different event for these, I'm just wondering why we don't usually need these at lower levels. Having two different events would complicate observations, as you would either need to add two observers (to detect either "change" or "modification") or you would have to fire modified event as well whenever you replaced the object.

Could you check how much it would complicate thing if you did not add the new event and just manage changes with the existing ones?